### PR TITLE
numpydocify Sankey.add.

### PR DESCRIPTION
**Pull request by anntzer**
Note that the previously stated default facecolor and linewidth only
apply in _internal.classic_mode, so don't need to be mentioned.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/sankey.py [hunk 1]
```diff
@@ -355,81 +355,84 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
...
-                            == 90, an *orientations* entry of 1 means to/from
-                            the left).
-          ===============   ===================================================
-
-        Valid kwargs are :meth:`matplotlib.patches.PathPatch` arguments:
+        Parameters
+        ----------
+        patchlabel : str
+            Label to be placed at the center of the diagram.
+            Note that *label* (not *patchlabel*) will be passed to the patch
```
Comment (timhoffm) [inline lib/matplotlib/sankey.py]: Simpler: Note that label (not patchlabel) can be passed as kwarg to create an entry in the legend.

File: lib/matplotlib/sankey.py [hunk 2]
```diff
@@ -355,81 +355,84 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
...
+            through ``**kwargs`` and can be used to create an entry in the
+            legend.
+
+        flows : list of float
+            Array of flow values.  By convention, inputs are positive and
+            outputs are negative.
+
+            The flows are placed along the top of the diagram from the inside
+            out in order of their index within the *flows* list or array.  They
```
Comment (timhoffm) [inline lib/matplotlib/sankey.py]: Simpler: ...within flows.

File: lib/matplotlib/sankey.py [hunk 3]
```diff
@@ -355,81 +355,84 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
...
+        labels : list of (str or None)
+            List of labels for the flows (or a single label to be used for all
+            flows).  Each label may be *None* (no label), or a labeling string.
+            If an entry is a (possibly empty) string, then the quantity for the
+            corresponding flow will be shown below the string.  However, if
+            the *unit* of the main diagram is None, then quantities are never
+            shown, regardless of the value of this argument.
+
+        trunklength : float
+            Length between the bases of the input and output groups.
```
Comment (timhoffm) [inline lib/matplotlib/sankey.py]: Can you give a unit?

File: lib/matplotlib/sankey.py [hunk 4]
```diff
@@ -355,81 +355,84 @@ def add(self, patchlabel='', flows=None, orientations=None, labels='',
...
+
+        Returns
+        -------
+        Sankey
+            The current `.Sankey` instance.
+
+        Other Parameters
+        ----------------
+        **kwargs
+          Additional keyword arguments set `matplotlib.patches.PathPatch`
```
Comment (timhoffm) [inline lib/matplotlib/sankey.py]: This should have 4-space indentation.

**Comment by anntzer** [comment]
Good to merge? :)

**Comment by timhoffm** [comment]
Oh, yes. I‘m always thinking of two reviews. But this is just documentation, for which one review is sufficient.

**Pull request merged**